pkg/provider/gitlab/fake/fake.go

```diff
@@ -17,20 +17,20 @@ import (
 	gitlab "github.com/xanzy/go-gitlab"
 )
 
-type GitlabMockClient struct {
+type GitlabMockProjectClient struct {
 	getVariable   func(pid interface{}, key string, options ...gitlab.RequestOptionFunc) (*gitlab.ProjectVariable, *gitlab.Response, error)
 	listVariables func(pid interface{}, options ...gitlab.RequestOptionFunc) ([]*gitlab.ProjectVariable, *gitlab.Response, error)
 }
 
-func (mc *GitlabMockClient) GetVariable(pid interface{}, key string, opt *gitlab.GetProjectVariableOptions, options ...gitlab.RequestOptionFunc) (*gitlab.ProjectVariable, *gitlab.Response, error) {
+func (mc *GitlabMockProjectClient) GetVariable(pid interface{}, key string, opt *gitlab.GetProjectVariableOptions, options ...gitlab.RequestOptionFunc) (*gitlab.ProjectVariable, *gitlab.Response, error) {
```

Comment:
some remarks:
- is there now a "group gitlabclient" and a "project gitlab" client ?
- how to you handle "enviroment_scope" which is not supported in group variables ? who do you handle "project_id" - is project = group ?

Reply:
- The client is just an internal abstraction of the gitlab library.
- Environment_scope is filtered "manually" (there is also a comment that the gitlab API does not support this yet)
- 
-- "only group id" -> error message because projectId is mandatory
-- "only project id" -> no change in behavior
-- group's from left to right (IDs in configuration) followed by project

Reply:
thx for the info @domizei385

Reply:
> (1) gitlab provider configured only with group-client
(2) gitlab provider configured only with project-client (status quo)
(3) gitlab provider configured with both (what is the merge strategy then?)

My GitLab knowledge is a bit rusty, IIRC:
1. the group vars are inherited but the API doesn't show them when calling `GET /projects/:id/variables`
2. If a project var `foo` is defined it takes precedence over a group-level variable

I would follow that order of precedence and implement the same behaviour.

From a user perspective it should be completely fine to pull only group variables without specifying a project. I think we should make the projectID optional and validate that any one of `project-id` or `group-id` has been configured. 
If both are configured we follow the same precedence: `project vars > group vars`.

Reply:
@moolen Thanks for the review. I fully agree. This expectation is exactly what I implemented.